internal/handlers/common/update.go

```diff
@@ -118,5 +107,102 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 		}
 	}
 
-	return true, nil
+	return changed, nil
+}
+
+// ValidateUpdateOperators validates update statement.
+func ValidateUpdateOperators(update *types.Document) error {
+	var err error
+	if err = checkAllModifiersSupported(update); err != nil {
+		return err
+	}
+	inc, err := extractValueFromUpdateOperator("$inc", update)
+	if err != nil {
+		return err
+	}
+	set, err := extractValueFromUpdateOperator("$set", update)
+	if err != nil {
+		return err
+	}
+	_, err = extractValueFromUpdateOperator("$setOnInsert", update)
+	if err != nil {
+		return err
+	}
+	if err = checkConflictingChanges(set, inc); err != nil {
+		return err
+	}
+	return nil
+}
+
+// checkAllModifiersSupported checks that update document contains only modifiers that are supported.
+func checkAllModifiersSupported(update *types.Document) error {
+	for _, updateOp := range update.Keys() {
+		switch updateOp {
+		case "$inc":
+			fallthrough
+		case "$set":
+			fallthrough
+		case "$setOnInsert":
+			// supported
+		default:
+			return NewWriteErrorMsg(
+				ErrFailedToParse,
+				fmt.Sprintf(
+					"Unknown modifier: %s. Expected a valid update modifier or pipeline-style "+
+						"update specified as an array", updateOp,
+				))
+		}
+	}
+	return nil
+}
+
+// checkConflictingChanges checks if there are the same keys in these documents and returns an error, if any.
+func checkConflictingChanges(a, b *types.Document) error {
+	if a == nil {
+		return nil
+	}
+	if b == nil {
+		return nil
+	}
+
+	for _, key := range a.Keys() {
+		if b.Has(key) {
+			return NewWriteErrorMsg(
+				ErrConflictingUpdateOperators,
+				fmt.Sprintf(
+					"Updating the path '%[1]s' would create a conflict at '%[1]s'", key,
+				),
+			)
+		}
+	}
+	return nil
+}
+
+// extractValueFromUpdateOperator gets operator `op` value and returns WriteError error if it is not a document.
+// For example, for update document
+// ```go
+// bson.D{
+// 	{"$set", bson.D{{"foo", int32(12)}}},
+// 	{"$inc", bson.D{{"foo", int32(1)}}},
+// 	{"$setOnInsert", bson.D{{"value", math.NaN()}}},
+// },
+// ```
+// The result returned for $setOnInsert operator is `bson.D{{"value", math.NaN()}`.
+func extractValueFromUpdateOperator(op string, update *types.Document) (*types.Document, error) {
```

Comment:
@rumyantseva please see! I've added the comment and renamed the function.

Reply:
godoc does not support Markdown syntax for code block. Here how it looks:

![image](https://user-images.githubusercontent.com/11512/171466778-44367ada-7553-4f07-8d87-6d4bb6d5f361.png)

The code block should be just indented to be rendered correctly

Reply:
removed quotes.

Reply:
updated godoc.
![Uploading Screenshot from 2022-06-02 13-59-36.png…]()